improver/calibration/dz_rescaling.py

```diff
@@ -424,8 +443,15 @@ def process(self, forecast: Cube, scaled_dz: Cube) -> Cube:
         fp_constr = self._create_forecast_period_constraint(forecast, scaled_dz)
         frt_constr = self._create_forecast_reference_time_constraint(forecast)
 
-        scaled_dz = scaled_dz.extract(fp_constr & frt_constr)
-        if not scaled_dz:
+        scaled_dz_extracted = scaled_dz.extract(fp_constr & frt_constr)
+
+        if not scaled_dz_extracted:
```

Comment:
```suggestion
        if not scaled_dz_extracted and self.frt_hour_leniency != 0:
```
I think that it is impossible for this block to find a match if a match hasn't already been found and the leniency is zero.